### PR TITLE
states: add e2e tests

### PR DESCRIPTION
**Pull request by fippo**
and fix misc bugs and terminology

### CODE REVIEW

**Comment by fippo** [comment]
@alvestrand @henbos there is an interesting thing about this sample that I only noticed when writing the test:
signaling state and ice connection state get a state change to "new/stable" (from "new/stabe"), connectionState does not.
That seems inconsistent...

**Comment by henbos** [comment]
Yeah I think we should calculate connectionState the same way we calculate iceConnectionState, in blink, so that it matches the JS exposed transport states.

But incidentally connectionState still exposes the ”failed” heuristic that was removed from iceConnectionState.

My kingdom for an end-of-candidates

**Comment by fippo** [comment]
ah... turns out there was an extra call adding the 'new' / 'stable' state to the div. Not a spec issue, a sample issue. Updated the sample and the test.

**Review by KaptenJansson — APPROVED** [review]

File: src/content/peerconnection/states/js/test.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+/*
+ *  Copyright (c) 2015 The WebRTC project authors. All Rights Reserved.
```
Comment (KaptenJansson) [inline src/content/peerconnection/states/js/test.js]: Update copyright year?

**Review by KaptenJansson — APPROVED** [review]

**Pull request merged**